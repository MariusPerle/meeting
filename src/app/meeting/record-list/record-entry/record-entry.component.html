<h2>
  {{ record().name }}
  <div class="date">
    {{ record().date | date }}
  </div>
</h2>

<table class="table">
  <thead>
  <tr>
    <th scope="col">Member</th>
    <th scope="col">Notes</th>
    <th scope="col" class="status">
      @if (totalTime) {
        Time
      } @else {
        Status
      }
    </th>
  </tr>
  </thead>
  <tbody>
    @for (member of record().members; track member.id; let even = $even) {
      <daily-record-member [recordMember]="member" [even]="even"></daily-record-member>
    }
    @if (totalTime) {
      <tr [ngClass]="{even: !(record().members.length % 2)}">
        <th scope="row"></th>
        <td></td>
        <td class="status">{{ totalTime | date: 'mm:ss' }}</td>
        <td></td>
      </tr>
    }
  </tbody>
</table>
