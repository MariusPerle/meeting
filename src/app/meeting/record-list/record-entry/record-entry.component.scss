:host {
    h2 {
        margin-top: 2rem;
        text-align: center;

        .date {
            font-size: medium;
        }
    }

    table {
        .status {
            width: 10ch;
            text-align: end;
        }

        daily-record-member {
            display: contents;
        }

        .even th, .even td {
            background: rgba(242,242,242,0.77);
        }
    }
}