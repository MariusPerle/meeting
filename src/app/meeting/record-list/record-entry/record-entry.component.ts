import { Component, input } from '@angular/core';
import { CommonModule } from '@angular/common';
import { RecordMemberComponent } from './record-member/record-member.component';
import { Record } from '../../+state/utils/meeting.interface';

@Component({
    selector: 'daily-record-entry',
    imports: [CommonModule, RecordMemberComponent],
    templateUrl: './record-entry.component.html',
    styleUrl: './record-entry.component.scss',
})
export class RecordEntryComponent {
    record = input.required<Record>();

    get totalTime() {
        return this.record().members.reduce(
            (acc, member) => acc + member.time,
            0,
        );
    }
}
