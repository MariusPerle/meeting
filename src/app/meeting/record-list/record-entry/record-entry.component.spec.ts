import { ComponentFixture, TestBed } from '@angular/core/testing';
import { RecordEntryComponent } from './record-entry.component';
import {
    recordFactory,
    recordMemberFactory,
} from '../../+state/utils/meeting-mock';
import { provideMockStore } from '@ngrx/store/testing';

describe('RecordEntryComponent', () => {
    let fixture: ComponentFixture<RecordEntryComponent>;

    beforeEach(async () => {
        await TestBed.configureTestingModule({
            imports: [RecordEntryComponent],
            providers: [provideMockStore()],
        }).compileComponents();

        fixture = TestBed.createComponent(RecordEntryComponent);
        fixture.componentRef.setInput(
            'record',
            recordFactory({
                id: '1',
                members: [
                    recordMemberFactory({ id: '2', time: 100000 }),
                    recordMemberFactory({ id: '3', time: 150000 }),
                ],
            }),
        );
        fixture.detectChanges();
    });

    it('should render correctly', () => {
        expect(fixture).toMatchSnapshot();
    });
});
