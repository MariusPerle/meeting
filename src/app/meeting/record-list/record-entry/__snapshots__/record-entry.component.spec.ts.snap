// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecordEntryComponent should render correctly 1`] = `
<meeting-record-entry
  record={[Function Function]}
>
  <h2>
     Meeting 
    <div
      class="date"
    >
       Jan 1, 2022 
    </div>
  </h2><table
    class="table"
  >
    <thead>
      <tr>
        <th
          scope="col"
        >
          Member
        </th>
        <th
          scope="col"
        >
          Notes
        </th>
        <th
          class="status"
          scope="col"
        >
           Time 
        </th>
      </tr>
    </thead>
    <tbody>
      <meeting-record-member>
        <tr
          class="even"
        >
          <th
            scope="row"
          >
          </th>
          <td>
          </td>
          <td
            class="status"
          >
             01:40 
          </td>
          <td
            class="styling-column"
          />
        </tr>
      </meeting-record-member>
      <meeting-record-member>
        <tr>
          <th
            scope="row"
          >
          </th>
          <td>
          </td>
          <td
            class="status"
          >
             02:30 
          </td>
          <td
            class="styling-column"
          />
        </tr>
      </meeting-record-member>
      <tr
        class="even"
      >
        <th
          scope="row"
        />
        <td />
        <td
          class="status"
        >
          04:10
        </td>
        <td />
      </tr>
    </tbody>
  </table>
</meeting-record-entry>
`;
